presto-docs/src/main/sphinx/security/salesforce.rst

```diff
@@ -0,0 +1,78 @@
+============================
+Salesforce Authentication
+============================
+
+Presto can be configured to enable frontend password authentication over
+HTTPS for clients, such as the CLI, or the JDBC and ODBC drivers. The
+username and password (or password and :ref:`security token` concatenation)
+are validated by having the Presto coordinator perform a login to Salesforce.
+
+This allows Presto admins to enable users to authenticate to Presto via their Salesforce
+basic credentials. This is *not* a Salesforce connector, and does not allow users to query
+Salesforce data. It's simply a means by which users can authenticate to Presto, similar
+to :doc:`ldap` or :doc:`password-file`.
+
+Salesforce Authenticator Configuration
+--------------------------------------
+
+Enable Salesforce authentication by creating an
+``etc/password-authenticator.properties`` file on the coordinator:
+
+.. code-block:: none
+
+    password-authenticator.name=salesforce
+    salesforce.allowed-organizations=allowed salesforce org id's, or the word all
+
+The following configuration properties are available:
+
+====================================   ==============================================
+Property                               Description
+====================================   ==============================================
+``salesforce.allowed-organizations``   Comma separated list of 18 character Salesforce.com
+                                       Organization ID's for a second simple layer of security.
+                                       This option can be explicitly ignored using ``all``
+                                       here, which will bypass any check of the authenticated
+                                       user's Salesforce Organization Id.
+
+``salesforce.cache-size``              Maximum size of the cache that holds authenticated users.
+                                       Defaults to ``4096``.
+
+``salesforce.cache-expire-duration``   Expire duration for an entry in cache since last write.
```

Comment:
whats the unit here?

Reply:
Also reword description ... maybe something like

```
Duration for an entry to remain the cache after the last write before it is removed.
```

Reply:
Will update the phrase.  

The type is `io.airlift.units.Duration`, and the unit is in the form of [this pattern](https://github.com/airlift/units/blob/master/src/main/java/io/airlift/units/Duration.java#L39), where the unit string should match one of [these cases](https://github.com/airlift/units/blob/master/src/main/java/io/airlift/units/Duration.java#L215).  It's similar to [`file.refresh-period`](https://prestosql.io/docs/current/security/password-file.html#password-authenticator-configuration)

Reply:
okay .. we probably should document that somewhere separately